src/vendor/wordpress/styles/_core-blocks.scss

```diff
@@ -115,7 +115,7 @@ $wp-button-gap: size.$spacing-gap-button-group-default;
  */
 .wp-block-code,
 .wp-block-jetpack-markdown pre,
-.legacy-post pre {
+.legacy-post :not(.c-comment) > :not(.c-comment__content) > pre {
```

Comment:
I'm trying to make sure I understand this:

1. Target `pre` elements
2. That are the direct child of an element that's _not_ `.c-comment__content`
3. As long as that parent element is _not_ the direct child of `.c-comment`

A couple of questions:

1. Is that third rule necessary? Could this line be switched to `.legacy-post :not(.c-comment__content) > pre`?
2. Will our legacy posts always have `pre` elements nested deeply enough for this to work? (a.k.a will there always be two layers of elements in between `.legacy-post` and the `pre` we want to style?

Reply:
1. Good point — I can drop the `:not(.c-comment) >` part of the selector.

2. Yeah, should be — `legacy-post` is applied to the outermost `o-container__content`, and `pre` elements will either be part of a blog post or a comment, and in both cases nested deeper in the DOM from that point.

Reply:
Cool, I just had one other thought about a potential edge case. Would this break if the `pre` was nested deeper within a comment. For example, imagine the following comment:

```md
> You wrote this code:
>
> ```js
> console.log('boop');
> ```
>
> I think you meant `beep`
```

In this scenario the `pre` would not be a direct child of `.c-comment__content` so wouldn't get the correct styles.

This is a total edge case but it makes me wonder if it would be safer to just un-set the relevant styles when a `pre` is in a comment?

```css
.c-comment pre {
  margin-inline: 0;
  padding-inline: 0;
}
```

Reply:
☹️ Yeah, good call. I'll update.